### ISSUE TITLE
Redirecting JavaScript console log outputs from JupyterLab

### ISSUE THREAD

**Issue by hbcarlos** [jupyterlab/jupyterlab#8877]
We want to redirect the JavaScript console logs from JupyterLab to an extension, the idea is to use it to debug other extensions without opening the browser's development tools because we have some cases where we don't have access to the development tools.

When one extension throws an exception or prints an error we are not aware of these errors without opening the development tools. The idea is to have a log console in JupyterLab for all the JavaScript errors.

### Problem

We built a console log extension that redirects the js logs and shows them, the problem is that this extension only redirects the js logs that are printed once the extension is loaded. For example, if there are errors loading other extensions like matplotlib, these errors are not captured by our extension because is not loaded yet.

### Proposed Solution

Adding priorities to load the extensions.

**Comment by krassowski** [comment]
I believe this extension is: https://github.com/QuantStack/jupyterlab-js-logs. I think that it would immensely help with error reporting for https://github.com/jupyterlab/jupyterlab_app. Are you still maintaining the extension/are there plans to move it upstream?

**Comment by hbcarlos** [comment]
Yes, we are actively maintaining the extension. What do you mean for upstream, move it to jupyterlab core extensions or the jupyterlab org?

In any case, both options are fine for me.

**Comment by krassowski** [comment]
Right, sorry, I saw the mention of Lab 2.0 and did not see the PyPI install but it's just the readme that was not converted to follow the 3.0 cookiecutter.

I was thinking about including it as a part of the core distribution to reduce the complexity of error reporting for all non-web-savy users of JupyterLab (i.e. so I could edit the bug reporting template and instead of pointing to "how to open the web dev tools in 5 most common browsers and then navigate to console in 3 different inspector implementations" just say that you can find the logs in here and there in our GUI), so if there are no technical arguments against it (e.g. browser support) I would advocate for moving straight to the core extensions.

However if the larger team decides that it is preferable not to have it in the core, I don't mind if it is in QS org or in the Lab org as long as we could include it in jupyterlab-app.

**Comment by hbcarlos** [comment]
It's okay for me. Can we maybe bring it to the next lab meeting and get some feedback?

**Comment by jtpio** [comment]
It would indeed be great to have it in core. 

This is also something that was partly discussed in https://github.com/jupyterlab/jupyterlab/issues/9735, whether we could reuse the existing log console for all kind of log message.

**Comment by jtpio** [comment]
https://github.com/jupyterlab/jupyterlab/issues/9735 has been closed but I'm not sure the current log console allows for other plugins to write to it?

So it looks like we have at least two options:

1. Bring `jupyterlab-js-logs` to core as-is
2. Update the log console to allow for other extensions to depend on it and add custom messages.

Probably option number 2 would be more generic and scalable in the long term? This would also avoid having two log console implementations in core, and two user facing consoles.

**Comment by blink1073** [comment]
The current log console can be used by other plugins.  Both the [`debugger`](https://github.com/jupyterlab/jupyterlab/blob/ceeb8e344ce58dac1f9ae7b9c54ae981a0466cad/packages/debugger-extension/src/index.ts#L623) and [`notebook`](https://github.com/jupyterlab/jupyterlab/blob/d48e0c04efb786561137fb20773fc15788507f0a/packages/notebook-extension/src/nboutput.ts#L45) extensions are using it.

**Comment by RDeRenzi** [comment]
Sorry to interfere in a discussion among professionals.

I posted 1 month ago the [following message ](https://stackoverflow.com/questions/73425350/move-notebook-log-console-in-jupyter-lab )on on stackoverflow:

_I am using a notebook as a single data analysis gui interface by means of ipywidget. Using the widgets performs tasks and self updates the gui. The text output is a cronological log of the data analysis. For this reason a good, although unusual target for the log would be the log console. However the standard "down" location of Log console is inconvenient because of the browser aspect ratio, with limited vertical space and lots of horizontal space._

Whatever the use of the log console (mine is perhaps just another one). there are certainly instances when the aspect ratio of the bottom window (very large and very shallow) does not help. It would be useful to be albe le to move it in a side-by-side configuration with the notebook one is debugging (or getting text logs from). This can be done for instance for two notebooks, by tab drag&drop, but does not work from the down part of the lab (please tell me if I'm wrong). 

Other very important feature that make the log desirable for my purpose are that it updates, its view is anchored at the last message, and  it can be scrolled). Not so easy to get with e.g. redirecting output to another notebook (does not update automatically), to a pipe and from there  to a terminal (my old solution)...  

Thanks for consideration
